### ISSUE TITLE
[BUG] Compatibility with React v19

### ISSUE THREAD

**Issue by CHE1RON** [juliencrn/usehooks-ts#602]
### Describe the bug

Currently, when using React v19 (currently RC), it gives an error regarding `ref` objects being passed to hooks, see here:

![image](https://github.com/juliencrn/usehooks-ts/assets/122524301/193a5194-6f76-4c53-826c-510bace940c2)

### To Reproduce

Simply follow the [upgrade guide](https://react.dev/blog/2024/04/25/react-19-upgrade-guide) and install React v19 RC, then try to build the project using `usehooks-ts`.

### Expected behavior

`build` step succeeds

**Comment by fstaffa** [comment]
@CHE1RON I was trying to replicate the problem but I was unable to. Can you confirm it is not caused by https://react.dev/blog/2024/04/25/react-19-upgrade-guide#ref-cleanup-required ?

**Comment by CHE1RON** [comment]
thanks for letting me know, I'll confirm that and get back to you!

**Comment by aaronmw** [comment]
Same problem here. Had to cast :(

```typescript
const containerElementRef = useRef<HTMLDivElement>(null)
const isHovering = useHover(containerElementRef as RefObject<HTMLDivElement>)
//                                              ^ 😭 
```

**Comment by rshackleton** [comment]
> Same problem here. Had to cast :(
> 
> ```ts
> const containerElementRef = useRef<HTMLDivElement>(null)
> const isHovering = useHover(containerElementRef as RefObject<HTMLDivElement>)
> //                                              ^ 😭 
> ```

I'd sooner annotate with `@ts-expect-error` so the cast can be removed once the type error is resolved.

```ts
const containerElementRef = useRef<HTMLDivElement>(null)
// @ts-expect-error React 19 type compatibility, nullable ref can be ignored.
const isHovering = useHover(containerElementRef as RefObject<HTMLDivElement>)
//                                              ^ 😭 
```

**Comment by rtrembecky** [comment]
encountered the same problem with `useResizeObserver`. also outlined e.g. here: https://github.com/ZeeCoder/use-resize-observer/issues/108

**Comment by thany** [comment]
Same happens for `useOnClickOutside`.

**Comment by iwan-uschka** [comment]
https://github.com/juliencrn/usehooks-ts/pull/665

**Comment by juliencrn** [comment]
I’m closing this issue as usehooks now supports React 19. Feel free to reopen it if any issues persist or new ones arise.
Apologies for the delay!

**Comment by mzizka** [comment]
The nullable ref type problem is not the same thing as React 19 compatibility (#650), although it is specific to React 19. The issue persists with the latest version of the package, and you have to work around it by casting or ignoring the error, as outined above.

**Comment by iwan-uschka** [comment]
👆 https://github.com/juliencrn/usehooks-ts/pull/665#issuecomment-2639164217

**Comment by lachieh** [comment]
Yeah, sorry @juliencrn. Could you re-open this? This is one that is going to have to be resolved with a major version bump.
